### PR TITLE
Feature | Stats page

### PR DESCRIPTION
**Pull request by nipunh**
Stats Page Implementation

## Description
Added a new route, created hook and frontend for stats page.

## Linked Issues
Feature | Stats page #233

## Type of Change
- New feature (non-breaking change which adds functionality)

## Changes

- Add new route to backend /stats
- created a new hook to get stats data
- created frontend to show stats

## Checklist before requesting a review

- [ *] I have performed a self-review of my code
- [ *] I assure there are no similar/duplicate pull requests regarding the same issue
- [ *] My changes follow the project's coding guidelines and best practices
- [ *] Code is formatted properly and lint check passes successfully
- [ ] I have made corresponding changes to the documentation (if applicable)

## Additional Notes

Additional Feature :  Graph for tracking blogs per month

### CODE REVIEW

**Comment by aadeshkulkarni** [comment]
build fails

**Comment by nipunh** [comment]
> build fails

Should i push changes for config files? How can i check the reason for build failure. I cannot access vercel.

**Comment by aadeshkulkarni** [comment]
> > build fails
> 
> 
> 
> Should i push changes for config files? How can i check the reason for build failure. I cannot access vercel.

npm run build

**Comment by nipunh** [comment]
> > > build fails
> > 
> > 
> > Should i push changes for config files? How can i check the reason for build failure. I cannot access vercel.
> 
> npm run build

Works on local system, but failing here

File: frontend/src/pages/Stats.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            }
+        }} />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+
+export default Stats
```
Comment (aadeshkulkarni) [inline frontend/src/pages/Stats.tsx:249]: 1. Code is not formatted properly. Please use linting to fix it.
2. Please create stats folders in component and put all auxiliary components inside it.
3. Please attach a video recording of your final stats page. (Light & Dark mode, desktop & Mobile)

Comment (nipunh) [reply frontend/src/pages/Stats.tsx:249]: Will do

**Comment by nipunh** [comment]
Screenshot 2024-07-22 at 5 43 15 PM
Screenshot 2024-07-22 at 5 43 26 PM
Screenshot 2024-07-22 at 5 54 20 PM
Screenshot 2024-07-22 at 5 54 29 PM
Screenshot 2024-07-22 at 5 54 53 PM

DO let me know if any other changes are required

**Comment by aadeshkulkarni** [comment]
Everything looks good. 
Once you resolve the merge conflict, we can merge.

**Comment by nipunh** [comment]
> Everything looks good. Once you resolve the merge conflict, we can merge.

Hey, we can merge it

**Comment by nipunh** [comment]
> Everything looks good. Once you resolve the merge conflict, we can merge.

@aadeshkulkarni Hey, we can merge it?

**Comment by aadeshkulkarni** [comment]
Thank you for your contribution! :)

**Pull request merged**